main/__tests__/Rating.test.tsx

```diff
@@ -69,5 +69,29 @@ describe('[Rating] render', () => {
       expect(rendered).toMatchSnapshot();
       expect(rendered).toBeTruthy();
     });
+
+    it('should simulate customItem and containerStyle props', (): void => {
+      const rendered = renderer.create(
+        component({
+          ...defaultProps,
+          customItem: {
+            onComponent: <View />,
+            offComponent: <View />,
+          },
+          containerStyle: {
+            width: 300,
+          },
+          testID: 'RATING_ID',
+        }),
+      );
+
+      rendered.update(component({ value: 3 }));
```

Comment:
This updated component's `total` property is undefined. So it seems that this method updates the `total` value to `undefined`. And the snapshot shows that the wrapper has only 1 child component. (Because `new Array(undefined)` makes [ undefined, ] and its length is 1)
 I think that's not really what you want. 
And it seems to some other parts of using the 'update' method have the same problem.

Reply:
Thanks. I will change this as `{...defaultProps, value: 3}` in another PR.